cpu/cc2538/radio/cc2538_rf_radio_ops.c

```diff
@@ -236,30 +268,48 @@ static int _confirm_cca(ieee802154_dev_t *dev)
 {
     (void) dev;
 
-    RFCORE_XREG_RFIRQM0 |= RXPKTDONE;
+    int res = -EAGAIN;
+    _disable_irq();
+    assert(cc2538_state == CC2538_STATE_CCA || cc2538_state == CC2538_STATE_CONFIRM_CCA);
+
+    if (cc2538_state != CC2538_STATE_CONFIRM_CCA) {
+        goto end;
+    }
 
-    return cc2538_cca_status;
+    cc2538_state = CC2538_STATE_READY;
+    RFCORE_XREG_FRMCTRL0 &= ~CC2538_FRMCTRL0_RX_MODE_DIS;
+    res = cc2538_cca_status;
```

Comment:
What's happening here?
Please leave a comment.

Reply:
I added a couple of functions to enable and disable RX chain. Now it should be more clear